packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -109,7 +109,26 @@ export class DropdownBase extends BaseComponent<IDropdownInternalProps, IDropdow
   public componentWillReceiveProps(newProps: IDropdownProps): void {
     // In controlled component usage where selectedKey is provided, update the selectedIndex
     // state if the key or options change.
-    const selectedKeyProp: keyof IDropdownProps = this.props.multiSelect ? 'selectedKeys' : 'selectedKey';
+    let selectedKeyProp: 'defaultSelectedKeys' | 'selectedKeys' | 'defaultSelectedKey' | 'selectedKey';
+
+    // this does a shallow compare (assumes options are pure), for the purposes of determining whether
```

Comment:
I'm not sure this is the right approach--it assumes a lot about the consumer's implementation. We should at least compare array length in addition to reference equality. That seems like it would catch the most common cases without compromising perf. (Then if someone complains we can consider doing a more comprehensive comparison.)

Reply:
Pure component is a thing in React, and it's all the more common now with Redux. 

I believe we do the same thing in DetailsList, and this is besides a super edge case. That said, it's probably harmless to check the length as well. I'll make sure to add some documentation to the props as well to encourage purity.

Reply:
Actually length check is probably not the right thing to do here... you can change the options without changing the length.

Unless the length check is an OR condition..

Reply:
I meant `newProps.options !== this.props.options || newProps.options.length !== this.props.options.length`...though come to think of it, if they're modifying the array in place, the length will already have changed..... :headpalm: So maybe just stick with the check you have now.

Reply:
Could you sign off then? :)

Reply:
> I believe we do the same thing in DetailsList, and this is besides a super edge case. That said, it's probably harmless to check the length as well. I'll make sure to add some documentation to the props as well to encourage purity.

Specifically here: https://github.com/OfficeDev/office-ui-fabric-react/blob/738e270892f99957aecf567e4b107f8e4cf86176/packages/office-ui-fabric-react/src/components/List/List.tsx#L334

Which is to blame for a few duplicate issues, one of which currently tracked is https://github.com/OfficeDev/office-ui-fabric-react/issues/3607#issuecomment-443037062